marlowe-playground-client/src/Marlowe/Holes.purs

```diff
@@ -256,23 +253,40 @@ constructMarloweType constructorName (MarloweHole { row, column }) m = case Map.
       let
         newArgs = getMarloweConstructors marloweType
 
-        hole = MarloweHole { name, marloweType, row: 0, column: 0 }
+        hole = MarloweHole { name, marloweType, range: emptyRange }
       in
         constructMarloweType name hole newArgs
 
   parens 1 1 text = text
 
   parens _ _ text = "(" <> text <> ")"
 
-mkHole :: forall a. String -> { row :: Pos, column :: Pos } -> Term a
-mkHole name position = Hole name Proxy position
+type Range
+  = { startLineNumber :: Int
+    , startColumn :: Int
+    , endLineNumber :: Int
+    , endColumn :: Int
+    }
 
-mkDefaultTerm :: forall a. a -> Term a
-mkDefaultTerm a = Term a { row: 0, column: 0 }
+emptyRange :: Range
+emptyRange =
+  { startLineNumber: 0
+  , startColumn: 0
+  , endLineNumber: 0
+  , endColumn: 0
+  }
+
+-- We need to compare the fields in the correct order
+compareRange :: Range -> Range -> Ordering
+compareRange =
+  compare `on` _.startColumn
+    <> compare `on` _.startLineNumber
+    <> compare `on` _.endLineNumber
+    <> compare `on` _.endColumn
```

Comment:
Also worth considering:

```purescript
compareRange =
  fold
    [ compare `on` _.startColumn
    , compare `on` _.startLineNumber
    , compare `on` _.endLineNumber
    , compare `on` _.endColumn
    ]
```

Reply:
beauty is in the eye of the beholder - I'll stick with the current I think